src/Utils/samples.jl

```diff
@@ -185,35 +164,47 @@ function sample(X::LazySet{N}; kwargs...) where {N<:Real}
 end
 
 """
-    Sampler{S<:LazySet, D<:Distribution}
+    Sampler
+
+Abstract type for defining new sample methods.
+
+### Notes
+
+All subtypes should implement a `_sample_from_sampler!` method.
+"""
+abstract type Sampler end
+
+"""
+    RejectionSampler{S<:LazySet, D<:Distribution} <: Sampler
 
-Type used for Rejections Sampling from an arbitrary `LazySet`.
+Type used for rejection sampling of an arbitrary `LazySet` `X`.
 
 ### Fields
 
 - `X`           -- lazyset
 - `box_approx`  -- Distribution from which the sample is drawn
 """
-struct Sampler{S<:LazySet, D<:Distribution}
+struct RejectionSampler{S<:LazySet, D<:Distribution} <:Sampler
     X::S
     box_approx::Vector{D}
 end
 
-function Sampler(X, distribution=Uniform)
-    box = _canonical_length(X)
-    box_approx = [distribution(box[i,:]...) for i = 1:size(box,1)]
-    return Sampler(X, box_approx)
+function RejectionSampler(X, distribution=Uniform)
+    B = box_approximation(X)
+    canonical_support =  hcat(low(B), high(B))
+    dims = size(canonical_support,1)
+    box_approx = [distribution(canonical_support[i,:]...) for i = 1:dims]
+    return RejectionSampler(X, box_approx)
 end
 
 
 """
-    _rejection_sampling!(D::Vector{Vector{N}},
-           sampler::Sampler;
+    _sample_from_sampler!(D::Vector{Vector{N}},
+           sampler::RejectionSampler;
            rng::AbstractRNG=GLOBAL_RNG,
            seed::Union{Int, Nothing}=nothing) where {N<:Real}
 
-Rejection sampling from a box approximation of a set and distribution
-defined in `sampler`.
+Sampling points from a `Sampler`.
```

Comment:
I think the documentation should say what this specific method does.
```suggestion
Sample points using rejection sampling.
```